src/core/gps/qgsgpstoolsinterface.h

```diff
@@ -0,0 +1,62 @@
+/***************************************************************************
+    qgsgpstoolsinterface.h
+    ---------------------
+    begin                : May 2025
+    copyright            : (C) 2025 by Germán Carrillo
+    email                : german at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSGPSTOOLSINTERFACE_H
+#define QGSGPSTOOLSINTERFACE_H
+
+#include <QObject>
+
+class QgsGpsConnection;
+class QgsLineSymbol;
+
+/**
+ * \ingroup core
+ * \class QgsGpsToolsInterface
+ * \brief Abstract interface class for the QGIS GPS tools.
+ * \since QGIS 3.44
+ */
+class CORE_EXPORT QgsGpsToolsInterface : public QObject
```

Comment:
At this stage this doesn't need to be a QObject, let's avoid:

```suggestion
class CORE_EXPORT QgsGpsToolsInterface
```